src/main/java/org/opentripplanner/routing/core/RoutingRequest.java

```diff
@@ -718,6 +722,33 @@ public void setDateTime(Date dateTime) {
 
     public void setDateTime(String date, String time, TimeZone tz) {
         Date dateObject = DateUtils.toDate(date, time, tz);
+        if (dateObject == null) {
+            // Handle bad date time.
+            if (invalidDateStrategy == null) {
+                throw new IllegalArgumentException("Date or time parameter is invalid.");
+            } else {
+                // Fix the date with the provided strategy.
+                LOG.warn("Could not parse date/time. Attempting invalid date strategy: {}", invalidDateStrategy);
+                switch (invalidDateStrategy.toUpperCase()) {
+                    case "USE_CURRENT":
+                        // Attempt to use provided time.
+                        Date now = new Date();
+                        date = new SimpleDateFormat("yyyy-MM-dd").format(now);
+                        dateObject = DateUtils.toDate(date, time, tz);
+                        if (dateObject == null) {
+                            // Time didn't parse. Use current time instead.
+                            LOG.warn("Couldn't parse time. Using current time instead.");
+                            dateObject = now;
+                        }
+                        break;
+                    // TODO: Add other strategies? For example, guess the nearest date to the one provided.
+                    default:
+                        // If invalidDateStrategy is not one of the above
+                        throw new IllegalArgumentException("Date or time parameter is invalid.");
```

Comment:
No context info is given in the error message (values), hence leaving the user in the dark. When I tried to run this code locally, this message was not returned to the client or logged - so changing it to something better would not help much.